packages/azure/data_stream/auditlogs/elasticsearch/ingest_pipeline/default.yml

```
@@ -229,6 +229,35 @@ processors:
   - set:
       field: event.kind
       value: event
+  - append:
+      field: related.entity
+      value: '{{{ azure.auditlogs.properties.initiated_by.user.id }}}'
+      allow_duplicates: false
+      if: ctx.azure?.auditlogs?.properties?.initiated_by?.user?.id != null && ctx.azure.auditlogs.properties.initiated_by.user.id != ''
+  - append:
+      field: related.entity
+      value: '{{{ azure.auditlogs.properties.initiated_by.app.servicePrincipalId }}}'
+      allow_duplicates: false
+      if: ctx.azure?.auditlogs?.properties?.initiated_by?.app?.servicePrincipalId != null && ctx.azure.auditlogs.properties.initiated_by.app.servicePrincipalId != ''
+  - script:
+      description: Appends target resources identifiers to `related.entity`
+      lang: painless
+      source: |
+        ctx.related = ctx.related ?: [:];
+        ctx.related.entity = ctx.related.entity ?: [];
+        if (ctx.azure.auditlogs.properties.target_resources != null) {
+            for (String i : ctx.azure.auditlogs.properties.target_resources.keySet()) {
+                def resource = ctx.azure.auditlogs.properties.target_resources[i];
```

```suggestion
            for (String k : ctx.azure.auditlogs.properties.target_resources.keySet()) {
                def resource = ctx.azure.auditlogs.properties.target_resources[k];
```
or perhaps
```suggestion
            for (String id : ctx.azure.auditlogs.properties.target_resources.keySet()) {
                def resource = ctx.azure.auditlogs.properties.target_resources[id];
```
if the key is an ID (it's not clear to me that this is the case).

`i` is an unconventional name for a non-integer variable.

---

indeed. thanks!